keystorage/hashicorp/src/main/java/tech/pegasys/signing/hashicorp/HashicorpConnection.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.signing.hashicorp;
+
+import static io.vertx.core.http.HttpMethod.GET;
+
+import tech.pegasys.signing.hashicorp.config.KeyDefinition;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import javax.net.ssl.SSLException;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientResponse;
+
+public class HashicorpConnection {
+
+  private static final String ERROR_HTTP_CLIENT_CALL =
+      "Error while waiting for response from Hashicorp Vault";
+
+  private static final String DEFAULT_HASHICORP_KEY_NAME = "value";
+
+  private final HttpClient httpClient;
+  private final long requestTimeoutMs;
+
+  public HashicorpConnection(final HttpClient httpClient, final long requestTimeoutMs) {
+    this.httpClient = httpClient;
+    this.requestTimeoutMs = requestTimeoutMs;
+  }
+
+  public String fetchKey(final KeyDefinition key) {
```

Comment:
This name is bit confusing to me, as it's not fetching the key is fetching the value for the value? Perhaps just fetch would be sufficient